packages/task/src/common/task-protocol.ts

```diff
@@ -22,9 +22,14 @@ export const TaskServer = Symbol('TaskServer');
 export const TaskClient = Symbol('TaskClient');
 
 export interface TaskConfiguration {
-    readonly type: string;
-    /** A label that uniquely identifies a task configuration */
+    /**
+     * Source of the task configuration.
+     * For a configured task, it is the name of the root folder, while for a provided task, it is the name of the provider.
+     */
+    readonly source: string;
```

Comment:
It worth to also mention it in the Task extension's [readme file](https://github.com/theia-ide/theia/tree/master/packages/task)

Reply:
the readme has been updated. I edited the section for extension contributions, as `source` does not need to be defined in `tasks.json`.